File: Exemplos/App_eXperience_XamarinAndroid/E1_Xamarin_Android_M8-font/XamarinAndroid-M8-master/Balanca/BalancaPage.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace M8
{
    [Activity(Label = "BalancaPage")]
    public class BalancaPage : Activity
    {
        private static Context mContext;

        public Balanca balanca;

        TextView textReturnValueBalanca;

        RadioGroup radioGroupModels;
        RadioButton radioButtonDP30CK;

        Spinner spinnerProtocols;

        Button buttonConfigurarBalanca;
        Button buttonLerPeso;

        string typeModel = "DP30CK";
        string typeProtocol = "PROTOCOL 0";

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Create your application here
            SetContentView(Resource.Layout.balanca_page);

            mContext = this;

            balanca = new Balanca(this);

            textReturnValueBalanca = FindViewById<TextView>(Resource.Id.textReturnValueBalanca);
            radioButtonDP30CK = FindViewById<RadioButton>(Resource.Id.radioButtonDP30CK);
            spinnerProtocols = FindViewById<Spinner>(Resource.Id.spinnerProtocols);
            buttonConfigurarBalanca = FindViewById<Button>(Resource.Id.buttonConfigurarBalanca);
            buttonLerPeso = FindViewById<Button>(Resource.Id.buttonLerPeso);

            //CONFIGS MODEL BALANÇA
            radioButtonDP30CK.Checked = true;
            radioGroupModels = FindViewById<RadioGroup>(Resource.Id.radioGroupModels);
            
            radioGroupModels.CheckedChange += (s, e) =>
            {
                switch (e.CheckedId)
                {
                    case Resource.Id.radioButtonDP30CK:
                        typeModel = "DP30CK";
                        break;
                    case Resource.Id.radioButtonSA110:
                        typeModel = "SA110";
                        break;
                    case Resource.Id.radioButtonDPSC:
                        typeModel = "DPSC";
                        break;
                }
            };

            spinnerProtocols.SetSelection(0);

            //CONFIGS PROTOCOLS
            spinnerProtocols.ItemSelected += (s, e) =>
            {
                typeProtocol = spinnerProtocols.SelectedItem.ToString();
            };

            buttonConfigurarBalanca.Click += delegate
            {
                Dictionary<string, Object> mapValues = new Dictionary<string, Object>();

                mapValues.Add("model", typeModel);
                mapValues.Add("protocol", typeProtocol);

                balanca.ConfigBalanca(mapValues);
            };

            buttonLerPeso.Click += delegate
            {
                string retorno = balanca.LerPesoBalanca();
                textReturnValueBalanca.Text = retorno;
            };

        }

        public static void Alert(String titleAlert, String message)
        {
            Android.App.AlertDialog AlertDialog = new Android.App.AlertDialog.Builder(mContext).Create();
            AlertDialog.SetTitle(titleAlert);
            AlertDialog.SetMessage(message);
            AlertDialog.SetButton("OK", delegate
            {
                AlertDialog.Dismiss();
            });
            AlertDialog.Show();
        }
    }
}

File: Exemplos/App_eXperience_Kotlin/Elgin_AppExperience_Kotlin-font/app/src/main/java/com/example/e1_kotlin_r/BalancaPage.kt
package com.example.e1_kotlin_r

import android.os.Bundle
import android.view.View
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.appcompat.app.AppCompatActivity
import com.example.e1_kotlin_r.Services.Balanca
import java.util.*


class BalancaPage : AppCompatActivity() {
    private var balanca: Balanca? = null
    private var textReturnValueBalanca: TextView? = null
    private lateinit var radioGroupModels: RadioGroup
    private lateinit var radioButtonDP30CK: RadioButton
    private lateinit var spinnerProtocols: Spinner
    private lateinit var buttonConfigurarBalanca: Button
    private lateinit var buttonLerPeso: Button
    private var typeModel = "DP30CK"
    var typeProtocol = "PROTOCOL 0"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_balanca_page)
        balanca = Balanca(this)
        textReturnValueBalanca = findViewById(R.id.textReturnValueBalanca)
        radioButtonDP30CK = findViewById(R.id.radioButtonDP30CK)
        spinnerProtocols = findViewById(R.id.spinnerProtocols)
        buttonConfigurarBalanca = findViewById(R.id.buttonConfigurarBalanca)
        buttonLerPeso = findViewById(R.id.buttonLerPeso)


        //CONFIGS MODEL BALANÇA
        radioButtonDP30CK.isChecked = true
        radioGroupModels = findViewById(R.id.radioGroupModels)
        radioGroupModels.setOnCheckedChangeListener { _, checkedId ->
            when (checkedId) {
                R.id.radioButtonDP30CK -> typeModel = "DP30CK"
                R.id.radioButtonSA110 -> typeModel = "SA110"
                R.id.radioButtonDPSC -> typeModel = "DPSC"
            }
        }

        //CONFIGS PROTOCOLS
        spinnerProtocols.onItemSelectedListener = object : OnItemSelectedListener {
            override fun onNothingSelected(parent: AdapterView<*>?) {}
            override fun onItemSelected(adapter: AdapterView<*>, v: View, i: Int, lng: Long) {
                typeProtocol = adapter.getItemAtPosition(i).toString()
            }
        }
        buttonConfigurarBalanca.setOnClickListener { sendConfigBalanca() }
        buttonLerPeso.setOnClickListener { sendLerPesoBalanca() }
    }

    private fun sendConfigBalanca() {
        val mapValues: MutableMap<String?, Any?> = HashMap()
        mapValues["model"] = typeModel
        mapValues["protocol"] = typeProtocol
        balanca!!.configBalanca(mapValues)
    }

    private fun sendLerPesoBalanca() {
        val retorno = balanca!!.lerPesoBalanca()
        textReturnValueBalanca!!.text = retorno
    }
}